### PR TITLE
Avoid reviewing own PRs by auto-merging with bot token

### PR DESCRIPTION
**Pull request by philipbalinov**
Use `mondoo-mergebot` app token when approving/merging PRs in order to avoid scenarios where we are attempting to create and approve a PR with the same user.

### CODE REVIEW

File: .github/workflows/pr-test-lint.yml [hunk 1]
```diff
@@ -135,13 +135,19 @@ jobs:
...
-        run: |
-          gh pr review ${{ steps.pr.outputs.number }} --approve
-          gh pr merge ${{ steps.pr.outputs.number }} --squash --merge
-        env:
-          GH_TOKEN: ${{secrets.GITHUB_TOKEN}}
-
+      # fetch a token for the mondoo-mergebot app
+      - name: Generate token
+        id: generate-token
+        uses: tibdex/github-app-token@v1
```
Comment (czunker) [inline .github/workflows/pr-test-lint.yml]: There is now an official action for this:
https://github.com/actions/create-github-app-token

We should switch to this one. It also has a more recent release.

File: .github/workflows/pr-test-lint.yml [hunk 2]
```diff
@@ -135,13 +135,19 @@ jobs:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
           filterOutClosed: true
           filterOutDraft: true
-      - name: Approve and merge a PR
-        run: |
-          gh pr review ${{ steps.pr.outputs.number }} --approve
```
Comment (czunker) [inline .github/workflows/pr-test-lint.yml:140]: Sorry, I missed this one on the first review:
We need to approve the PR. Or is one of the actions taking care of this?

**Review by czunker — APPROVED** [review]
Thanks @philipbalinov

File: .github/workflows/pr-test-lint.yml (hunk 1 above)
Comment (philipbalinov) [reply .github/workflows/pr-test-lint.yml]: switched

File: .github/workflows/pr-test-lint.yml (hunk 2 above)
Comment (philipbalinov) [reply .github/workflows/pr-test-lint.yml:140]: reverted back to the manual `gh pr` executions

**Pull request merged**